lib/livebook/notebook/explore/intro_to_livebook.livemd

```diff
@@ -60,6 +60,32 @@ for _ <- 1..3 do
 end
 ```
 
+<!-- livebook:{"parent_index":2} -->
+
+## Branching sections
+
+Additionally, you can make a section **branch out** from any
+previous regular section. Hover over the section name to reveal
+available actions and click on the branch icon to select the
+parent section.
+
+You still have access to all the previous data:
+
+```elixir
+{message, cats}
+```
+
+The important characteristic of a branching section is that
```

Comment:
I think we should mention that:

1. They are separate processes
2. The process dictionary is copied
3. All bindings are copied

And therefore there may be a cost related to this operation. However, I am worried then this is no longer "intro to livebook" material and maybe we should move it to the Elixir + Livebook one. Thoughts?

Reply:
These are quite specific implementation details and I'm not sure if it's something the user should be concerned with. From their perspective a branching section should have the same scope as if it was regular section below the parent (2. and 3.) and it is concurrent (1.).

I wasn't sure where to place this, but this seems more Livebook related and also the introduction talks about `Sections`, so this kinda convinced me we should show branching right there.

Reply:
Cool, this sounds good!

Reply:
WDYT?

![image](https://user-images.githubusercontent.com/17034772/125808701-576209d4-0178-4d85-b7e4-9b9e1d3d15b1.png)

Reply:
Beautiful!